src/target/riscv/riscv.c

```diff
@@ -666,6 +669,33 @@ static int set_trigger(struct target *target, unsigned int idx, riscv_reg_t tdat
 	return ERROR_OK;
 }
 
+static int sdtrig_disable_trigger(struct target *target, unsigned int idx)
+{
+	// Select which trigger to use
+	if (riscv_reg_set(target, GDB_REGNO_TSELECT, idx) != ERROR_OK)
+		return ERROR_FAIL;
+
+	// Disable the trigger by writing 0 to it
+	if (riscv_reg_set(target, GDB_REGNO_TDATA1, 0) != ERROR_OK)
+		return ERROR_FAIL;
+
+	return ERROR_OK;
+}
+
+static int sdtrig_set_t1(struct target *target, unsigned int idx,
+		riscv_reg_t tdata1, riscv_reg_t tdata1_ignore_mask)
+{
+	return stdtrig_set_tx_impl(target, idx, tdata1, /* tdata2 */ 0,
+			tdata1_ignore_mask, /* update tdata2 */ false);
+}
+
+static int sdtrig_set_t1t2(struct target *target, unsigned int idx,
+		riscv_reg_t tdata1, riscv_reg_t tdata2, riscv_reg_t tdata1_ignore_mask)
+{
+	return stdtrig_set_tx_impl(target, idx, tdata1, tdata2, tdata1_ignore_mask,
+			/* update tdata2 */ true);
+}
+
```

Comment:
I'd recommend to rename the functions to say `tdata1`, `tdata2` instead of `t1`, `t2`. The reason is that the `tN` convention is already used to denote trigger types in `maybe_add_trigger_tN`.

Possible names could be:

- `sdtrig_set_t1()` --> `sdtrig_set_tdata1()`
- `sdtrig_set_t1t2()` --> `sdtrig_set_tdata1_2()`
- `stdtrig_set_tx_impl()` --> `stdtrig_set_tdata_impl()`

Or you can come up with another naming convention; my point is just to have different names for `maybe_add_trigger_*()` and `sdtrig_set_*()`.

Reply:
I like the naming scheme you've proposed. Will try to use it.

As for `maybe_add_trigger_`  - I don't like this inconsistency too.. But I think it would be worth to move all trigger-related code to a separate file and do rename during this activity.